velox/serializers/PrestoSerializer.cpp

```diff
@@ -1747,6 +1838,11 @@ void PrestoVectorSerde::deserialize(
   bool useLosslessTimestamp = options != nullptr
       ? static_cast<const PrestoOptions*>(options)->useLosslessTimestamp
       : false;
+  std::unique_ptr<folly::io::Codec> codec = options == nullptr
```

Comment:
nit: we can have an anonymous function which takes options ptr to return the codec
```
std::unique_ptr<folly::io::Codec> getCompressionCodec(options* );
```